lib/plugins/aws/package/compile/events/apiGateway/lib/deployment.js

```diff
@@ -13,7 +13,7 @@ module.exports = {
         Type: 'AWS::ApiGateway::Deployment',
         Properties: {
           RestApiId: this.provider.getApiGatewayRestApiId(),
-          StageName: this.provider.getStage(),
+          StageName: this.provider.serverless.service.provider.nodeEnv,
```

Comment:
I think this is not correct. You have to keep `getStage()` here because that is the method that returns the stage unambiguously (also in 3rd party plugins and other locations). This introduces incompatibilities and removes the SPOT (single point of truth) for the stage name. Additionally using "nodeEnv" is not runtime agnostic - Node.js is only one of many runtimes used and available.

Reply:
Oh this was not supposed to be included in the Pull request. I was also using my fork for my custom naming needs. Forgot that even after you open a PR, that any commits to that branch gets updated in the PR. Im going to close this PR and create a different branch for the changes in the first commit. Thanks for pointing this out